src/libraries/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1979,6 +2004,14 @@ private int InternalGetHashCode(int hashCode)
             return hashCode & Lower31BitMask;
         }
 
+#if BIT64
```

Comment:
I believe that BIT64 works in CoreLib only. Libraries do not have infrastructure for bitness-specific ifdefs.

Reply:
Oh! Nuts. You're right.  

What do you think of moving HashSet to corelib? It looks like it would not bring a great deal: ISet.cs, HashsetEqualityComparer.cs, BitHelper.cs, and one string.

Or, does JIT inline across assemblies? If so could we call into corelib for GetFastModMultiplier and FastMod -- in 32 bit the multiplier would be passed redundantly and be ignored.

Or, we could abandon this, which would be a pity.

Reply:
I would be fine with moving the whole System.Collections as part of https://github.com/dotnet/runtime/issues/2138

Reply:
Any reason why we shouldn't move HashSet in this PR, as part of making this change? Seems like everything would come out in the wash.

Reply:
We like to avoid no-op refactoring with actual changes.

Reply:
What I should have said is  --  just move these 3 types, then follow up by making the change (or in the same PR and not squash).

Reply:
I can start on either one.  It would be better if I start on #2138 to get more familiar with the process.

Reply:
@danmosemsft  I believe the tests are not covering all constructors.